### ISSUE TITLE
make this dynamic with state prefix eg: state-on

### ISSUE THREAD

**Issue by JonasWijne** [royteusink/tailwindcss-state#1]
would be cool if this wouldn't need any configured states.
```html
<div class="state-on"> 
   <div class="state-on:bg-red-500"> ... </div>
</div>
```

**Comment by royteusink** [comment]
I don't think we can implement this at this moment. There is currently no `matchVariants` function in the core of tailwindcss. Only `matchUtilities` and `matchComponents` [[see source]](https://github.com/tailwindlabs/tailwindcss/blob/a7263a8f6faf989cb98553491d5c456d0b86de9b/src/lib/setupContextUtils.js#L304).

Something like this would be great:

```html
<div class="collapsed">
    <div class="state-[collapsed]:bg-red-500"> ... </div>
</div>
```

If someone has a solution for this, please make let me know.
